src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -314,11 +314,23 @@ public readonly void DisposePendingDisposablesOnException()
             Debug.Assert(Current.AsyncDisposable is null);
             DisposeFrame(Current.CollectionEnumerator, ref exception);
 
-            int stackSize = Math.Max(_count, _continuationCount);
-            for (int i = 0; i < stackSize - 1; i++)
+            if (_stack is not null)
             {
-                Debug.Assert(_stack[i].AsyncDisposable is null);
-                DisposeFrame(_stack[i].CollectionEnumerator, ref exception);
+                int currentIndex = _count - _indexOffset;
+                int stackSize = Math.Max(currentIndex, _continuationCount);
```

Comment:
I'd expect that `_indexOffset` would also be subtracted from `_continuationCount`.

Reply:
This stumped me for a while as well, but it does seem that the two values have an off-by-one relationship:

https://github.com/eiriktsarpalis/runtime/blob/dd4054029ba47f69fa16dcea228977e608963cb6/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs#L254